components/xprofile.php

```diff
@@ -96,176 +178,198 @@ public function list_fields_( $_, $assoc_args ) {
-				);
-				return array( 'success', $success );
+				return array( 'success', sprintf( 'Deleted XProfile field "%s" (ID %d)', $name, $id ) );
 			} else {
-				return array( 'error', "Failed deleting field $field_id." );
+				return array( 'error', sprintf( 'Failed deleting field %d.', $field_id ) );
 			}
 		} );
 	}
 
 	/**
+	 * Get an XProfile field.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <field-id>
+	 * : Identifier for the field.
+	 *
+	 * [--fields=<fields>]
+	 * : Limit the output to specific fields. Defaults to all fields.
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 *  ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - json
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp xprofile get_field 500
+	 *     $ wp bp xprofile get_field 56 --format=json
+	 *
+	 * @since 1.5.0
+	 */
+	public function get_field( $args, $assoc_args ) {
+		$field_id = $args[0];
+
+		if ( ! is_numeric( $field_id ) ) {
+			WP_CLI::error( 'Please provide a numeric field ID.' );
+		}
+
+		$object = xprofile_get_field( $field_id );
+
+		if ( is_object( $object ) && ! empty( $object->id ) ) {
+			$object_arr = get_object_vars( $object );
+			if ( empty( $assoc_args['fields'] ) ) {
+				$assoc_args['fields'] = array_keys( $object_arr );
+			}
+			$formatter = $this->get_formatter( $assoc_args );
+			$formatter->display_item( $object_arr );
+		} else {
+			WP_CLI::error( 'No field found.' );
+		}
+	}
+
+	/**
 	 * Set profile data for a user.
 	 *
 	 * ## OPTIONS
 	 *
-	 * --user_id=<user>
+	 * --user-id=<user>
 	 * : Identifier for the user. Accepts either a user_login or a numeric ID.
 	 *
-	 * --field_id=<field-id>
-	 * : Field ID. Accepts either the name of the field or a numeric ID.
+	 * --field-id=<field>
+	 * : Identifier for the field. Accepts either the name of the field or a numeric ID.
 	 *
 	 * --value=<value>
 	 * : Value to set.
 	 *
-	 * [--is_required=<is-required>]
-	 * : Whether a non-empty value is required. Default: false
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp xprofile set_data --user-id=45 --field-id=120 --value=teste
+	 *     $ wp bp xprofile set_data --user-id=user_test --field-id=445 --value=another_test
 	 *
 	 * @since 1.2.0
 	 */
 	public function set_data( $args, $assoc_args ) {
-		$r = wp_parse_args( $assoc_args, array(
-			'user_id'     => '',
-			'field_id'    => '',
-			'value'       => '',
-			'is_required' => false,
-		) );
-
-		$user = $this->get_user_id_from_identifier( $r['user_id'] );
+		$user = $this->get_user_id_from_identifier( $assoc_args['user-id'] );
 
 		if ( ! $user ) {
-			WP_CLI::error( 'No user found by that username or id' );
-			return;
+			WP_CLI::error( 'No user found by that username or ID' );
 		}
 
-		// Validate field
-		// We need this info anyway for the success message.
-		if ( ! is_numeric( $r['field_id'] ) ) {
-			$field_id = xprofile_get_field_id_from_name( $r['field_id'] );
-		} else {
-			$field_id = intval( $r['field_id'] );
+		if ( empty( $assoc_args['value'] ) ) {
```

Comment:
I think this check here is not needed as the value is already a required one. Right?!

Reply:
Correct. Fixing now.